src/main/kotlin/org/rust/ide/lineMarkers/RsMethodLineSeparatorProvider.kt

```diff
@@ -0,0 +1,51 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.lineMarkers
+
+import com.intellij.codeHighlighting.Pass
+import com.intellij.codeInsight.daemon.DaemonCodeAnalyzerSettings
+import com.intellij.codeInsight.daemon.LineMarkerInfo
+import com.intellij.codeInsight.daemon.LineMarkerProvider
+import com.intellij.openapi.editor.colors.CodeInsightColors
+import com.intellij.openapi.editor.colors.EditorColorsManager
+import com.intellij.openapi.editor.markup.GutterIconRenderer
+import com.intellij.openapi.editor.markup.SeparatorPlacement
+import com.intellij.openapi.util.text.StringUtil
+import com.intellij.psi.PsiElement
+import com.intellij.psi.util.PsiTreeUtil
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.ext.getPrevNonCommentSibling
+
+class RsMethodLineSeparatorProvider : LineMarkerProvider {
+
+    override fun getLineMarkerInfo(element: PsiElement): LineMarkerInfo<PsiElement>? {
+        if (DaemonCodeAnalyzerSettings.getInstance().SHOW_METHOD_SEPARATORS) {
+            if (element.canHaveSeparator()) {
+                val prevSibling = element.getPrevNonCommentSibling()
+                if (prevSibling.canHaveSeparator() && (element.wantsSeparator() || prevSibling.wantsSeparator())) {
+                    return createLineSeparatorByElement(element)
+                }
+            }
+        }
+        return null
+    }
+
+    private fun PsiElement?.canHaveSeparator() = this is RsFunction
+
+    private fun PsiElement?.wantsSeparator() = if (this == null) false else StringUtil.getLineBreakCount(text) > 0
```

Comment:
Are we sure that we don't want line separator for one line functions?
For example, kotlin doesn't show separators only for one-line single-expression functions like this
```kotlin
fun foo(x): Int = x * x
```
 But rust doesn't have such syntax at all, so maybe drop this condition?

Reply:
I think this condition is necessary for this case:
```rust
trait Foo {
    fn f1();
    fn f2();
    fn f3();
}
```